### PR TITLE
Add unit tests for removing agents functions in worker nodes

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR adds unit tests for the following functions:
* [`remove_bulk_agents`](https://github.com/wazuh/wazuh/blob/f5326f7c3684c77d012a6b103f5e70914e575f95/framework/wazuh/cluster/worker.py#L209)
* [`_check_removed_agents`](https://github.com/wazuh/wazuh/blob/f5326f7c3684c77d012a6b103f5e70914e575f95/framework/wazuh/cluster/worker.py#L272)

It also improves the implementation of `remove_bulk_agents` function to make it more maintainable.

Best regards,
Marta

### CODE REVIEW

**Comment by mgmacias95** [comment]
Hello,

I have added the fix of https://github.com/wazuh/wazuh/issues/2887 in this PR as well since it affects the way agents are removed.

Best regards,
Marta

File: framework/wazuh/utils.py [hunk 1]
```diff
@@ -764,9 +764,12 @@ def _parse_legacy_filters(self):
         """
         Parses legacy filters.
         """
-        legacy_filters_as_list = {name: value.split(',') if isinstance(value, unicode) or isinstance(value,str) else value for name, value in self.legacy_filters.items()}
-        self.query_filters += [{'value': None if subvalue == "null" else subvalue, 'field': '{}${}'.format(name,i), 'operator': '=', 'separator': 'OR' if len(value) > 1 else 'AND', 'level': 0}
-                               for name, value in legacy_filters_as_list.items() for subvalue,i in zip(value, range(len(value))) if not self._pass_filter(subvalue)]
+        legacy_filters_as_list = {name: value.split(',') if isinstance(value,str) else value for name, value in self.legacy_filters.items()}
+        self.query_filters += [{'value': None if subvalue == "null" else subvalue, 'field': '{}${}'.format(name,i),
```
Comment (crd1985) [inline framework/wazuh/utils.py]: We should consider the case when `value` is a number which doesn't have the len property. Let's change the unit tests to detect this potential error and ensure the correctness of the implementation :)

Comment (mgmacias95) [reply framework/wazuh/utils.py]: You were right, the filter failed when a non-string value was used as a filter value. Fixed in https://github.com/wazuh/wazuh/pull/2869/commits/280b61021171628836518d6cbd7039fd07e8dea0

**Comment by mgmacias95** [comment]
## Passed unit tests
```python
# /var/ossec/framework/python/bin/pytest GitHub/wazuh/framework/wazuh/tests/test_agent.py -vv
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 25 items

GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_default PASSED                                                                                                                                                                             [  4%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select0-all-None-0] PASSED                                                                                                                                                          [  8%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select1-all-None-1] PASSED                                                                                                                                                          [ 12%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select2-all-None-1] PASSED                                                                                                                                                          [ 16%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select3-Active,Pending-None-0] PASSED                                                                                                                                               [ 20%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select4-Disconnected-None-1] PASSED                                                                                                                                                 [ 24%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select5-Disconnected-1s-1] PASSED                                                                                                                                                   [ 28%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select6-Disconnected-2h-0] PASSED                                                                                                                                                   [ 32%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select7-all-15m-2] PASSED                                                                                                                                                           [ 36%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select8-Active-15m-0] PASSED                                                                                                                                                        [ 40%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select9-Active,Pending-15m-1] PASSED                                                                                                                                                [ 44%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.201] PASSED                                                                                                                                                              [ 48%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.202] PASSED                                                                                                                                                              [ 52%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.202;registerIP=any] PASSED                                                                                                                                               [ 56%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[status=Disconnected;lastKeepAlive>34m] PASSED                                                                                                                                        [ 60%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[(status=Active,status=Pending);lastKeepAlive>5m] PASSED                                                                                                                              [ 64%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search0-3] PASSED                                                                                                                                                                   [ 68%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search1-3] PASSED                                                                                                                                                                   [ 72%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search2-1] PASSED                                                                                                                                                                   [ 76%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search3-5] PASSED                                                                                                                                                                   [ 80%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search4-2] PASSED                                                                                                                                                                   [ 84%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[active-9m-1-None] PASSED                                                                                                                                                  [ 88%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[all-1s-5-None] PASSED                                                                                                                                                     [ 92%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[pending,neverconnected-30m-1-None] PASSED                                                                                                                                 [ 96%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[55-30m-0-1729] PASSED                                                                                                                                                     [100%]

============================================================================================================================= warnings summary ==============================================================================================================================
GitHub/wazuh/framework/wazuh/__init__.py:82
  /home/vagrant/GitHub/wazuh/framework/wazuh/__init__.py:82: DeprecationWarning: invalid escape sequence \w
    line_regex = re.compile('(^\w+)="(.+)"')

GitHub/wazuh/framework/wazuh/__init__.py:125
  /home/vagrant/GitHub/wazuh/framework/wazuh/__init__.py:125: DeprecationWarning: invalid escape sequence \w
    line_regex = re.compile('(^\w+)="(.+)"')

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================================================================================================================== 25 passed, 2 warnings in 0.66 seconds ===================================================================================================================
```
```python
# /var/ossec/framework/python/bin/pytest GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py -vv
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 11 items

GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344-agents_to_remove0] PASSED [  9%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n-agents_to_remove1] PASSED [ 18%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 !test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-agents_to_remove2] PASSED [ 27%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 !test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 !test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-agents_to_remove3] PASSED [ 36%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-\n-agents_to_remove4] PASSED [ 45%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[\n-001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-agents_to_remove5] PASSED [ 54%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n004 test4 any d7ae2f7fe182d202f9088ecb7a0f8899fee7f192c0c0d2d4db906dtfc22a7ad5-agents_to_remove6] PASSED [ 63%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n004 test4 any d7ae2f7fe182d202f9088ecb7a0f8899fee7f192c0c0d2d4db906dtfc22a7ad5-agents_to_remove7] PASSED [ 72%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 !test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n004 test4 any d7ae2f7fe182d202f9088ecb7a0f8899fee7f192c0c0d2d4db906dtfc22a7ad5-agents_to_remove8] PASSED [ 81%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_remove_bulk_agents[agents_to_remove0] PASSED                                                                                                                                                          [ 90%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_remove_bulk_agents[agents_to_remove1] PASSED                                                                                                                                                          [100%]

============================================================================================================================= warnings summary ==============================================================================================================================
GitHub/wazuh/framework/wazuh/configuration.py:279
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:279: DeprecationWarning: invalid escape sequence \s
    regex_comment = re.compile("^\s*#")

GitHub/wazuh/framework/wazuh/configuration.py:280
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:280: DeprecationWarning: invalid escape sequence \s
    regex_title = re.compile("^\s*\[(.*)\]\s*\[(.*)\]\s*\[(.*)\]\s*")

GitHub/wazuh/framework/wazuh/configuration.py:281
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:281: DeprecationWarning: invalid escape sequence \{
    regex_name_groups = re.compile("(\{\w+:\s+\S+\s*\S*\})")

GitHub/wazuh/framework/wazuh/configuration.py:282
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:282: DeprecationWarning: invalid escape sequence \s
    regex_check = re.compile("^\s*(\w:.+)")

GitHub/wazuh/framework/wazuh/configuration.py:283
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:283: DeprecationWarning: invalid escape sequence \s
    regex_var = re.compile("^\s*\$(\w+)=(.+)")

GitHub/wazuh/framework/wazuh/configuration.py:368
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:368: DeprecationWarning: invalid escape sequence \s
    regex_comment = re.compile("^\s*#")

GitHub/wazuh/framework/wazuh/configuration.py:369
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:369: DeprecationWarning: invalid escape sequence \s
    regex_check = re.compile("^\s*(.+)\s+!\s*(.+)\s*::\s*(.+)")

GitHub/wazuh/framework/wazuh/configuration.py:398
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:398: DeprecationWarning: invalid escape sequence \s
    regex_comment = re.compile("^\s*#")

GitHub/wazuh/framework/wazuh/configuration.py:399
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:399: DeprecationWarning: invalid escape sequence \s
    regex_check = re.compile("^\s*(.+)\s+!\s*(.+)\s*!\s*(.+)")

GitHub/wazuh/framework/wazuh/configuration.py:400
  /home/vagrant/GitHub/wazuh/framework/wazuh/configuration.py:400: DeprecationWarning: invalid escape sequence \s
    regex_binary_check = re.compile("^\s*(.+)\s+!\s*(.+)\s*!")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================================================================================== 11 passed, 10 warnings in 0.92 seconds ===================================================================================================================
```

## Snippet of the status filter working correctly
```javascript
# curl -u foo:bar "localhost:55000/agents?pretty&status=Active,Pending"
{
   "error": 0,
   "data": {
      "items": [
         {
            "os": {
               "arch": "x86_64",
               "codename": "Bionic Beaver",
               "major": "18",
               "minor": "04",
               "name": "Ubuntu",
               "platform": "ubuntu",
               "uname": "Linux |master |4.15.0-46-generic |#49-Ubuntu SMP Wed Feb 6 09:33:07 UTC 2019 |x86_64",
               "version": "18.04.2 LTS"
            },
            "id": "000",
            "status": "Active",
            "manager": "master",
            "lastKeepAlive": "9999-12-31 23:59:59",
            "node_name": "master",
            "dateAdd": "2019-03-25 11:18:07",
            "version": "Wazuh v3.9.0",
            "ip": "127.0.0.1",
            "registerIP": "127.0.0.1",
            "name": "master"
         },
         {
            "id": "001",
            "status": "Pending",
            "manager": "",
            "lastKeepAlive": "2019-03-25 11:30:04",
            "node_name": "unknown",
            "dateAdd": "2019-03-25 11:29:07",
            "ip": "any",
            "registerIP": "any",
            "name": "pepe"
         }
      ],
      "totalItems": 2
   }
}
root@master:/home/vagrant/GitHub/wazuh/framework# curl -u foo:bar "localhost:55000/agents?pretty&status=Active,Pending&older_than=3m"
{
   "error": 0,
   "data": {
      "items": [],
      "totalItems": 0
   }
}
# curl -u foo:bar "localhost:55000/agents?pretty&status=Active,Pending&older_than=10s"
{
   "error": 0,
   "data": {
      "items": [
         {
            "node_name": "unknown",
            "ip": "any",
            "lastKeepAlive": "2019-03-25 11:30:04",
            "manager": "",
            "status": "Pending",
            "id": "001",
            "registerIP": "any",
            "name": "pepe",
            "dateAdd": "2019-03-25 11:29:07"
         }
      ],
      "totalItems": 1
   }
}
```

**Comment by mgmacias95** [comment]
Done! Now the output of pytest is:
```python
# /var/ossec/framework/python/bin/pytest GitHub/wazuh/framework/wazuh/tests/test_agent.py -vv
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 25 items

GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_default PASSED                                                                                                                                                                             [  4%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select0-all-None-0] PASSED                                                                                                                                                          [  8%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select1-all-None-1] PASSED                                                                                                                                                          [ 12%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select2-all-None-1] PASSED                                                                                                                                                          [ 16%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select3-Active,Pending-None-0] PASSED                                                                                                                                               [ 20%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select4-Disconnected-None-1] PASSED                                                                                                                                                 [ 24%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select5-Disconnected-1s-1] PASSED                                                                                                                                                   [ 28%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select6-Disconnected-2h-0] PASSED                                                                                                                                                   [ 32%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select7-all-15m-2] PASSED                                                                                                                                                           [ 36%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select8-Active-15m-0] PASSED                                                                                                                                                        [ 40%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_select[select9-Active,Pending-15m-1] PASSED                                                                                                                                                [ 44%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.201] PASSED                                                                                                                                                              [ 48%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.202] PASSED                                                                                                                                                              [ 52%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.202;registerIP=any] PASSED                                                                                                                                               [ 56%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[status=Disconnected;lastKeepAlive>34m] PASSED                                                                                                                                        [ 60%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_query[(status=Active,status=Pending);lastKeepAlive>5m] PASSED                                                                                                                              [ 64%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search0-3] PASSED                                                                                                                                                                   [ 68%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search1-3] PASSED                                                                                                                                                                   [ 72%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search2-1] PASSED                                                                                                                                                                   [ 76%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search3-5] PASSED                                                                                                                                                                   [ 80%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_search[search4-2] PASSED                                                                                                                                                                   [ 84%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[active-9m-1-None] PASSED                                                                                                                                                  [ 88%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[all-1s-5-None] PASSED                                                                                                                                                     [ 92%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[pending,neverconnected-30m-1-None] PASSED                                                                                                                                 [ 96%]
GitHub/wazuh/framework/wazuh/tests/test_agent.py::test_get_agents_overview_status_olderthan[55-30m-0-1729] PASSED                                                                                                                                                     [100%]

========================================================================================================================= 25 passed in 0.52 seconds =========================================================================================================================
```
```python
# /var/ossec/framework/python/bin/pytest GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py -vv
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 11 items

GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344-agents_to_remove0] PASSED [  9%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n-agents_to_remove1] PASSED [ 18%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 !test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-agents_to_remove2] PASSED [ 27%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 !test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 !test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-agents_to_remove3] PASSED [ 36%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-\n-agents_to_remove4] PASSED [ 45%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[\n-001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-agents_to_remove5] PASSED [ 54%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n004 test4 any d7ae2f7fe182d202f9088ecb7a0f8899fee7f192c0c0d2d4db906dtfc22a7ad5-agents_to_remove6] PASSED [ 63%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n004 test4 any d7ae2f7fe182d202f9088ecb7a0f8899fee7f192c0c0d2d4db906dtfc22a7ad5-agents_to_remove7] PASSED [ 72%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_check_removed_agents[001 test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5-001 !test1 any 54cfda3bfcc817aadc8f317b3f05d676d174cdf893aa2f9ee2a302ef17ae6794\n002 test2 any 7a9c0990dadeca159c239a06031b04d462d6d28dd59628b41dc7e13cc4d3a344\n003 test3 any d7ae2f7fe182d202f9088ecb7a0c8899fee7f192c0c0d2d4db906d5fc22a7ad5\n004 test4 any d7ae2f7fe182d202f9088ecb7a0f8899fee7f192c0c0d2d4db906dtfc22a7ad5-agents_to_remove8] PASSED [ 81%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_remove_bulk_agents[agents_to_remove0] PASSED                                                                                                                                                          [ 90%]
GitHub/wazuh/framework/wazuh/cluster/tests/test_worker.py::test_remove_bulk_agents[agents_to_remove1] PASSED                                                                                                                                                          [100%]

========================================================================================================================= 11 passed in 0.71 seconds =========================================================================================================================
```

**Pull request merged**